.gitmodules

```diff
@@ -31,3 +31,12 @@
 [submodule "external/alfons"]
 	path = external/alfons
 	url = https://github.com/hjanetzek/alfons
+[submodule "external/curl"]
+	path = external/curl
+	url = https://github.com/lygstate/curl.git
+[submodule "external/zlib"]
+	path = external/zlib
+	url = https://github.com/madler/zlib.git
+[submodule "external/angle"]
+	path = external/angle
+	url = https://github.com/lygstate/angle.git
```

Comment:
@blair1618 one option to avoid fetching unneeded dependencies would be to checkout submodules explicitly using 
```
git submodule update --init external/zlib
git submodule update --init external/curl
git submodule update --init external/angle
```
This could be part of the ```$platform.cmake``` so that one only downloads the required submodules necessary for a given platform. 

@lygstate instead of using curl it would be good to use the Windows libraries - since for curl to be useful one also needs to build OpenSSL.